core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -228,18 +239,31 @@ public void testToQuery() throws IOException {
         assertLuceneQuery(testQuery, actualQuery, context);
     }
 
+    protected final Query createExpectedQuery(QB queryBuilder, QueryParseContext context) throws IOException {
+        Query expectedQuery = doCreateExpectedQuery(queryBuilder, context);
+        expectedQuery.setBoost(queryBuilder.boost());
+        return expectedQuery;
+    }
+
     /**
      * Creates the expected lucene query given the current {@link QueryBuilder} and {@link QueryParseContext}.
      * The returned query will be compared with the result of {@link QueryBuilder#toQuery(QueryParseContext)} to test its behaviour.
      */
-    protected abstract Query createExpectedQuery(QB queryBuilder, QueryParseContext context) throws IOException;
+    protected abstract Query doCreateExpectedQuery(QB queryBuilder, QueryParseContext context) throws IOException;
 
     /**
      * Run after default equality comparison between lucene expected query and result of {@link QueryBuilder#toQuery(QueryParseContext)}.
-     * Can contain additional assertions that are query specific. Empty default implementation.
+     * Can contain additional assertions that are query specific. Default implementation verifies that names queries are properly handled.
      */
-    protected void assertLuceneQuery(QB queryBuilder, Query query, QueryParseContext context) {
+    protected final void assertLuceneQuery(QB queryBuilder, Query query, QueryParseContext context) {
```

Comment:
I think it's a leftover cause some queries used to have different checks for query name, but that code is all gone now. will change